taskcluster/client.py

```
@@ -400,11 +400,23 @@ def _makeHttpRequest(self, method, route, payload):
                 )
 
             # Handle non 2xx status code and retry if possible
+            status = response.status_code
             try:
                 response.raise_for_status()
-                if response.status_code == 204:
+                if status == 204:
                     return None
 
+                # raise_for_status only catches status >= 400, but it also
+                # customizes the error message.  Let's use that customized
+                # error message for status >= 400, and craft one for other
+                # non-2xx statuses.
+                if status < 200 or status >= 300:
+                    raise requests.exceptions.RequestException(
```

Like the async, let's throw a taskcluster exception with these as properties.